network/src/compress.rs

```diff
@@ -116,3 +117,142 @@ pub fn compress(src: Bytes) -> Bytes {
 pub fn decompress(src: BytesMut) -> Result<Bytes, io::Error> {
     Message::from_compressed(src).decompress()
 }
+
+/// LengthDelimitedCodec with compress support
+pub struct LengthDelimitedCodecWithCompress {
+    length_delimited: length_delimited::LengthDelimitedCodec,
+    enable_compress: bool,
+    protocol_id: p2p::ProtocolId,
+}
+
+impl LengthDelimitedCodecWithCompress {
+    /// Create a new LengthDelimitedCodecWithCompress
+    pub fn new(
+        enable_compress: bool,
+        length_delimited: length_delimited::LengthDelimitedCodec,
+        protocol_id: p2p::ProtocolId,
+    ) -> Self {
+        Self {
+            length_delimited,
+            enable_compress,
+            protocol_id,
+        }
+    }
+}
+
+impl tokio_util::codec::Encoder<Bytes> for LengthDelimitedCodecWithCompress {
+    type Error = io::Error;
+    fn encode(&mut self, data: Bytes, dst: &mut BytesMut) -> Result<(), io::Error> {
+        let process = |data: &[u8], flag: u8, dst: &mut BytesMut| {
```

Comment:
Defining this logic as a helper method instead of an inline closure would make the code easier to read and reuse, and keeps the encode function more concise.

```suggestion
impl LengthDelimitedCodecWithCompress {
    fn process(&self, data: &[u8], flag: u8, dst: &mut BytesMut) -> { ... }
}
```